packit_service/worker/helpers/testing_farm.py

```diff
@@ -141,6 +141,285 @@ def parse_unknown_arguments(self, unknown_args: list[str]) -> None:
                 break
 
 
+class TestingFarmClient:
```

Comment:
Do you think it would make sense to take this out to a separate source file?

Reply:
I'm kinda for removing the `ServiceConfig` from the `TestingFarmClient` and pass directly API URL, token and ranch…

We should probably open at least an issue for that; I checked how is this client-logic handled in the TF CLI (to check if it's feasible to reuse it), but there are no abstractions :/

Reply:
> Do you think it would make sense to take this out to a separate source file?

Yes, I think so.

> I'm kinda for removing the `ServiceConfig` from the `TestingFarmClient` and pass directly API URL, token and ranch…

Makes sense if we separate it from the helpers.

Reply:
+1 for passing those directly, should make it more readable for usage of internal/public TF

Reply:
> Do you think it would make sense to take this out to a separate source file?

Where should it be? In `packit_service/worker/helpers`? That doesn't seem like a good fit. Perhaps directly in `packit_service`?

Reply:
to me, having it in helpers sounds ok, as it will still be used just by the TFHelper, right?